anton.baraiev/task_5/src/main/java/com/hillel/UserServlet.java

```diff
@@ -0,0 +1,136 @@
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        resp.setContentType("application/json");
+        Gson gson = new Gson();
+        CrudEventStatus status;
+        String message;
+        try{
+            CreateUserRequest createUserRequest = gson.fromJson(req.getReader(), CreateUserRequest.class);
+            User user = new User(createUserRequest.getName(), createUserRequest.getPhoneNumber(),
+                                    createUserRequest.getAge(), currentUserId);
+            users.put(currentUserId, user);
+            currentUserId++;
+            message = String.format("User %s was successfully added", createUserRequest.getName());
+            status = new CrudEventStatus(true, message);
+            log.info(message);
+        }catch (Exception e) {
+            message = "Exception while adding a user";
+            status= new CrudEventStatus(false, message);
+            log.error(message);
+            resp.setStatus(500);
+        }
+        resp.getWriter().println(status.getDescription());
+        resp.getWriter().flush();
+    }
+
+    @Override
+    protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        resp.setContentType("application/json");
+        Gson gson = new Gson();
+        CrudEventStatus status;
+        String message;
+        try{
+            CreateUserRequest createUserRequest = gson.fromJson(req.getReader(), CreateUserRequest.class);
+            int id = createUserRequest.getId();
+            if (users.containsKey(id)) {
+                User user = new User(createUserRequest.getName(), createUserRequest.getPhoneNumber(),
+                        createUserRequest.getAge(), id);
+                users.put(id, user);
+                message = String.format("User id = %d info was updated", id);
+                status = new CrudEventStatus(true, message);
+                log.info(message);
+            }else {
+                message = String.format("There is no user for id = %d", id);
+                status = new CrudEventStatus(false, message);
+                log.warn(message);
+                resp.setStatus(404);
+            }
+        }catch (Exception e) {
+            message = "Error, while trying to update user info";
+            status = new CrudEventStatus(false, message);
+            resp.setStatus(500);
+        }
+        resp.getWriter().println(status.getDescription());
+        resp.getWriter().flush();
+    }
+
+    @Override
+    protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        resp.setContentType("application/json");
+        String message;
+        CrudEventStatus status;
+        try{
+            if(req.getParameter("id") != null) {
+                int id = Integer.parseInt(req.getParameter("id"));
+                users.remove(id);
```

Comment:
You should check if user with given id exists. If not - return response with code 404